rasterio/_io.pxd

```diff
@@ -28,10 +28,11 @@ cdef class WarpedVRTReaderBase(DatasetReaderBase):
 
 cdef class InMemoryRaster:
     cdef GDALDatasetH _hds
-    cdef double transform[6]
+    cdef double gdal_transform[6]
     cdef int band_ids[1]
     cdef np.ndarray _image
     cdef object crs
+    cdef object transform
```

Comment:
@sgillies A comment here indicating that this is always an `Affine` might be helpful to avoid any ambiguity going forward (since there used to be a `transform` property of a different type...)